src/Common/src/Interop/Unix/libssl/Interop.libssl.cs

```diff
@@ -56,51 +56,48 @@ internal static partial class libssl
         internal static extern void SSL_set_accept_state(IntPtr ssl);
 
         [DllImport(Interop.Libraries.LibSsl)]
-        internal static extern int SSL_write(IntPtr ssl, IntPtr buf, int num);
+        internal static extern int SSL_write(SafeSslHandle ssl, IntPtr buf, int num);
 
         [DllImport(Interop.Libraries.LibSsl)]
-        internal static extern int SSL_read(IntPtr ssl, IntPtr buf, int num);
+        internal static extern int SSL_read(SafeSslHandle ssl, IntPtr buf, int num);
 
         [DllImport(Interop.Libraries.LibSsl)]
-        internal static extern int SSL_shutdown(IntPtr ssl);
-
-        [DllImport(Interop.Libraries.LibSsl)]
-        internal static extern IntPtr BIO_new(IntPtr type);
+        internal static extern int SSL_shutdown(SafeSslHandle ssl);
 
         [DllImport(Interop.Libraries.LibSsl)]
         internal static extern void SSL_set_bio(IntPtr ssl, IntPtr rbio, IntPtr wbio);
 
         [DllImport(Interop.Libraries.LibSsl)]
-        internal static extern int SSL_do_handshake(IntPtr ssl);
+        internal static extern int SSL_do_handshake(SafeSslHandle ssl);
 
         [DllImport(Interop.Libraries.LibSsl)]
-        internal static extern int SSL_state(IntPtr ssl);
+        internal static extern int SSL_state(SafeSslHandle ssl);
 
         [DllImport(Interop.Libraries.LibSsl)]
-        internal static extern int BIO_read(IntPtr bio, IntPtr buf, int num);
+        internal static extern int BIO_read(SafeBioHandle bio, IntPtr buf, int num);
 
         [DllImport(Interop.Libraries.LibSsl)]
-        internal static extern int BIO_write(IntPtr bio, IntPtr buf, int num);
+        internal static extern int BIO_write(SafeBioHandle bio, IntPtr buf, int num);
 
         [DllImport(Interop.Libraries.LibSsl)]
-        internal static extern IntPtr SSL_get_peer_certificate(IntPtr ssl);
+        internal static extern SafeX509Handle SSL_get_peer_certificate(SafeSslHandle ssl);
```

Comment:
I checked (again), and this method does require an X509_free call to be applied to the return value, so I'm very happy this is a SafeHandle'd method now :).